tools/publish/fetch-winbuild-artifacts.sh

```
@@ -0,0 +1,48 @@
+# Open one authenticated connection (prompts for the password once) and reuse
+# it for every ssh/scp call below, so the rest of the script runs unattended.
+CTRL_SOCK=$(mktemp -u)
+cleanup() { command ssh -S "$CTRL_SOCK" -O exit "$VM_USER@$VM_HOST" >/dev/null 2>&1 || true; }
+trap cleanup EXIT
+ssh -MNf -S "$CTRL_SOCK" "$VM_USER@$VM_HOST"
+ssh() { command ssh -S "$CTRL_SOCK" "$@"; }
+scp() { command scp -o ControlPath="$CTRL_SOCK" "$@"; }
+
+echo "Staging + zipping artifacts matching *$VERSION* (and RELEASES) under out/ on the VM..."
+scp "$LOCAL_PS1" "$VM_USER@$VM_HOST:$REMOTE_PS1"
+ssh "$VM_USER@$VM_HOST" "powershell -NoProfile -File $REMOTE_PS1 -Version $VERSION -OutRoot $VM_REPO/apps/harvester/out -ZipPath $REMOTE_ZIP"
```

Shellcheck (0.11.0)

[info] Note that, unescaped, this expands on the client side.

(SC2029)